neptune/alpha/internal/utils/json_file_splitter.py

```diff
@@ -14,50 +14,75 @@
 # limitations under the License.
 #
 import json
+from collections import deque
 from io import StringIO
 from json import JSONDecodeError
 from typing import Optional
 
 
 class JsonFileSplitter:
 
-    # TODO: experiment with larger buffer sizes
     BUFFER_SIZE = 64 * 1024
+    MAX_PART_READ = 8 * 1024
 
     def __init__(self, file_path: str):
         self._file = open(file_path, "r")
         self._decoder = json.JSONDecoder(strict=False)
-        self._buffer = StringIO()
+        self._part_buffer = StringIO()
+        self._part_read = 0
```

Comment:
Why do you have such a variable instead of using self._buffer.len ?

Reply:
Built-in `len` function does not work on StringIO and cannot found any documentation of `len` attribute.